packages/wrangler/e2e/dev-with-resources.test.ts

```diff
@@ -0,0 +1,606 @@
+import assert from "node:assert";
+import crypto from "node:crypto";
+import events from "node:events";
+import getPort from "get-port";
+import dedent from "ts-dedent";
+import { Agent, fetch } from "undici";
+import { beforeEach, describe, expect } from "vitest";
+import { WebSocket } from "ws";
+import { e2eTest } from "./helpers/e2e-wrangler-test";
+import { generateResourceName } from "./helpers/generate-resource-name";
+import { killAllWranglerDev, waitForReady } from "./helpers/wrangler";
+
+beforeEach(killAllWranglerDev);
```

Comment:
I think this is intending to kill any wrangler's running on your local machine, started even outside of the e2e test suite

Reply:
Right.
But equally, there might be wranglers left over by this test, so we would want be good citizens and clean these up too, no?

Reply:
No strong feeling either way, really. The slightly nice thing about using a `beforeEach()` is that it cleans up any hanging processes you have locally that would otherwise cause tests to fail before the first test runs

Reply:
Perhaps both?

Reply:
Changed to both